k8s/deploy-storageos/upgrade-helper/prepare-upgrade.sh

```diff
@@ -0,0 +1,257 @@
+
+    if [ -z "$image" ]; then
+        echo "Couldn't find the StorageOS installed version"
+        exit 1
+    fi
+
+    if [ "$image" != "$LATEST_VERSION" ]; then
+        if [ "$init_img" != "$LATEST_INIT" ]; then
+            echo "Current init container: $init_img"
+            print_green "Setting the image for the init container to $LATEST_INIT"
+            kubectl -n $ns patch ds/storageos-daemonset --type='merge' --patch "$(cat $INITCONTAINER_PATCH)"
+        fi
+        echo "Current StorageOS version: $image"
+        print_green "Setting image for DaemonSet storageos-daemonset to $LATEST_VERSION"
+        kubectl -n $ns set image ds/storageos-daemonset storageos=$LATEST_VERSION
+    fi
+}
+
+# list_pods_using_storageos_volumes Outputs the Pods using StorageOS PVCs
+list_pods_using_storageos_volumes() {
+    # TODO: assess if adding a dependency to JQ is ok
+
+    print_red "The following Pods are using StorageOS volumes."
+    print_red "They need to be scaled to 0 or be restarted after StorageOS is running the newer version."
+    print_red "Ensure that the StorageOS Pods are in READY state before you scale back up"
+
+    # Iterate over all the PVCs in the cluster that use the StorageOS
+    # provisioner and print any existing Pod that uses any of these PVCs. In
+    # practice, any Pod using a StorageOS volume
+    counter=0
+    while IFS= read -r line; do
+        local ns=$(cut -f1  -d' ' <(echo $line))
+        local pvc=$(cut -f2 -d' ' <(echo $line))
+
+        list=$(kubectl -n $ns get pod -ojson | jq -M --arg PVC "$pvc" -r '.items[]
+        | select(.spec.volumes[].persistentVolumeClaim.claimName != null)
+        | select(.spec.volumes[].persistentVolumeClaim.claimName == $PVC)
+        | { NameSpace: .metadata.namespace,  Pod: .metadata.name}')
+
+        if [ ! -z "$list" ]; then
+            echo "$list"
+            ((counter++))
+            list=''
+        fi
+
+    done < <(kubectl get pvc --all-namespaces -ojson \
+        | jq -r '.items[]
+        | select (.metadata.annotations."volume.beta.kubernetes.io/storage-provisioner"
+        | test("storageos"))
+        |  .metadata.namespace + " " + .metadata.name'
+    )
+
+    if [ $counter -eq 0 ]; then
+        print_green "No StorageOS Volumes are in use, you can proceed"
+    fi
+}
+
+####### MAIN #########
+
+
+verify_dependencies
+create_patches
+
+check_operator_version
+
+# Find current StorageOS installed version
+out=$(kubectl get pod --all-namespaces  -lapp=storageos,kind=daemonset --no-headers -ocustom-columns=_:.metadata.name,_:.metadata.namespace | head -1)
+
+pod="$(cut -d' ' -f1 <(echo $out))"
+ns="$(cut  -d' ' -f2 <(echo $out))"
+
+if [ -z "$pod" ]; then
+    echo "Couldn't find any StorageOS Pod"
+    exit 1
+fi
+
+set_updatestrategy_ondelete $ns
+set_rbac_rules_for_initcontainer $ns
+set_new_images $pod $ns
+
+list_pods_using_storageos_volumes
+
+echo "When you are ready, you can delete the StorageOS Pods to restart with the new version."
```

Comment:
I'd make the instructions here a bit more explicit since volumes will be unavailable during restart.  e.g.:

> To complete the upgrade you must restart the StorageOS Pods on each node.  While the container is being restarted, volumes accessed from application Pods on the node will be unavailable.  We recommend that you put the cluster in maintenance mode (link to docs) shutdown applications accessing StorageOS volumes on the node, and then delete the StorageOS Pod.  The DaemonSet will restart the Pod using the updated version.  Once all nodes have been upgraded, disable maintenance mode.

> It is safe to re-run this script.

Reply:
Just saw that you've got some of this text in the `list_pods_using_storageos_volumes` function.  Feel free to use whatever makes sense.

Reply:
It makes a lot of sense. Thank you. 
I put all the text in place as I think that reiterating is good and it gives a more clear view of what is happening.

It is also mentioned in the documentation that references the execution of the script.